### DISCUSSION TITLE
Authenticated Data Channels

### DISCUSSION THREAD

**Discussion by cjdsellers** [bmoscon/cryptofeed#414] [Q&A]
Hi Bryant,

Good work on starting to add the authenticated data channels support.

How can we find out which exchanges support authenticated channels, which are available, and which are a work in progress (intending to be supported)?

At the moment I can see a few examples, otherwise one would have to parse through the source code to find out?

Many thanks!

**Comment by bmoscon** [comment]
@cjdsellers - ok, thats a good question. Right now you'd need to call the info() method on the exchange class to find out what channels are supported, but thats cumbersome. I'll add documentation about supported data channels, including authenticated channels.

**Comment by bmoscon** [comment]
Here is the doc: https://github.com/bmoscon/cryptofeed/blob/master/docs/auth_channels.md

As you can see right now Gemini is the only supported exchange, and it only has a single auth'd channel. I'm currently adding support for Kraken, which has 2 auth'd channels

**Comment by cjdsellers** [comment]
> Here is the doc: https://github.com/bmoscon/cryptofeed/blob/master/docs/auth_channels.md
> 
> 
> 
> As you can see right now Gemini is the only supported exchange, and it only has a single auth'd channel. I'm currently adding support for Kraken, which has 2 auth'd channels

Hi Bryant,

Thanks for the update.

Are there plans to add account details/balances auth'd endpoint access?

My own use case is for trading platform integrations, minimally requiring auth'd order, trade, and account information. Probably at the very least order or trade/execution event streams, account details via REST would still work.

As part of such an integration there would no doubt be some contribution upstream to you to assist the implementation and maintenance of the adapter.

**Reply by bmoscon** [reply]
Kraken doesnt support those, but binance.us has a balance update. I'll add binance.us after kraken. Assuming that the binance US API is the same as their main exchange, that will in effect add support on binance as well.

**Reply by cjdsellers** [reply]
Excellent, I'll take another look at the library.

**Comment by walter211** [comment]
I want to add Binance support, is there any doc to guide?

**Reply by bmoscon** [reply]
no docs, but there was a PR: https://github.com/bmoscon/cryptofeed/pull/452

and a commit where I added gemini as an example: 
https://github.com/bmoscon/cryptofeed/commit/af2645e56de5f8b0b0d284009ac06b705494a178

**Comment by TSavo** [comment]
Is Binance.US now supported with authenticated channels?

**Comment by svabra** [comment]
Question: the [documentation](https://github.com/bmoscon/cryptofeed/blob/master/docs/auth_channels.md) says, there is no authentication for Binance. But why would people then implement the private channels for BALANCES and ORDER_INFO (for which one needs to be authenticated)? Just a question, no criticism.

**Reply by bmoscon** [reply]
the documentation is not always up to date or correct
